aws/resource_aws_security_group_attachment.go

```diff
@@ -0,0 +1,173 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSecurityGroupAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSecurityGroupAttachmentCreate,
+		Read:   resourceAwsSecurityGroupAttachmentRead,
+		Delete: resourceAwsSecurityGroupAttachmentDelete,
+		Schema: map[string]*schema.Schema{
+			"security_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network_interface_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsSecurityGroupAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	if err := attachSecurityGroupToInterface(d, meta); err != nil {
+		return err
+	}
+
+	return resourceAwsSecurityGroupAttachmentRead(d, meta)
+}
+
+func attachSecurityGroupToInterface(d *schema.ResourceData, meta interface{}) error {
+	sgID := d.Get("security_group_id").(string)
+	interfaceID := d.Get("network_interface_id").(string)
```

Comment:
It looks like we rely on this field, shouldn't it be `Required` in the schema then?